ecs-update/ecs-update.go

```diff
@@ -1798,13 +1937,85 @@ func fieldsYMLUseExternalECS(f *ast.File, ff fleetpkg.FieldsFile) (changed bool,
 		replacement := newNode(fmt.Sprintf("name: %s\nexternal: ecs", quoteKey(origField.Name)))
 
 		if err = p.ReplaceWithNode(f, replacement); err != nil {
-			return false, fmt.Errorf("faield to replace node: %w", err)
+			return false, fmt.Errorf("failed to replace node: %w", err)
 		}
 		changed = true
 	}
 	return changed, nil
 }
 
+// fieldsYMLDropExternalECS will find fields that can use 'external: ecs' and remove
+// their definitions. It will use the latest version of ECS to determine if the field
+// exists in ECS.
+func fieldsYMLDropExternalECS(f *ast.File, ff fleetpkg.FieldsFile) (changed bool, err error) {
+	// Use the flat representation so that we have access to the complete field name.
+	flatFields, err := fleetpkg.FlattenFields(ff.Fields)
+	if err != nil {
+		return false, err
+	}
+
+	for _, field := range flatFields {
+		// Does the field exist in ECS?
+		ecsField, _ := ecs.Lookup(field.Name, "")
+		if ecsField == nil {
+			continue
+		}
+
+		// The type must be the same in order to do the replacement.
```

Comment:
I think there are few additional attributes that are commonly overridden to modify the indexing behavior. If any of those exist then we would not be able to drop the definition. See

https://github.com/andrewkroh/fydler/blob/ad0a63f364ca24cfe5afc33607b34783946ef689/internal/analysis/useecs/useecs.go#L122-L133

Reply:
Will this need changing in the `fieldsYMLUseExternalECS` func? ISTM yes.

Reply:
Oh you right. I guess I discovered this problem after adding `-fields-yml-use-ecs` while tinkering with fydler.